lib/gcpspanner/chromium_daily_usage_stat_test.go

```diff
@@ -41,14 +41,162 @@ func TestListChromiumDailyUsageStatsForFeatureID(t *testing.T) {
+		time.Date(2000, time.January, 8, 0, 0, 0, 0, time.UTC),
+		5,
+		nil,
+	)
+
+	if !errors.Is(err, nil) {
+		t.Errorf("expected no error during listing of metrics. received %s", err.Error())
 	}
+	if token != nil {
+		t.Error("expected null token")
+	}
+	expectedStats := []ChromiumDailyUsageStatWithDate{
+		{
+			Date: civil.Date{
+				Year:  2000,
+				Month: time.January,
+				Day:   2,
+			},
+			Usage: big.NewRat(90, 100),
+		},
+		{
+			Date: civil.Date{
+				Year:  2000,
+				Month: time.January,
+				Day:   1,
+			},
+			Usage: big.NewRat(89, 100),
+		},
+	}
+
+	if !reflect.DeepEqual(expectedStats, stats) {
+		t.Errorf("unequal stats. expected (%+v) received (%+v) ", expectedStats, stats)
+	}
+}
+
+func testGetStatsPages(ctx context.Context, c *Client, t *testing.T) {
+	stats, token, err := c.ListChromiumDailyUsageStatsForFeatureID(
+		ctx,
+		"feature2",
+		time.Date(2000, time.January, 1, 0, 0, 0, 0, time.UTC),
+		time.Date(2000, time.December, 1, 0, 0, 0, 0, time.UTC),
+		2,
+		nil,
+	)
+
+	if !errors.Is(err, nil) {
+		t.Errorf("expected no error during listing of metrics. received %s", err.Error())
+	}
+	if token == nil {
+		t.Error("expected token")
+	}
+	expectedStats := []ChromiumDailyUsageStatWithDate{
+		{
+			Date: civil.Date{
+				Year:  2000,
+				Month: time.January,
+				Day:   15,
+			},
+			Usage: big.NewRat(91, 100),
+		},
+		{
+			Date: civil.Date{
+				Year:  2000,
+				Month: time.January,
+				Day:   2,
+			},
+			Usage: big.NewRat(90, 100),
+		},
+	}
+	if !reflect.DeepEqual(expectedStats, stats) {
+		t.Errorf("unequal stats. expected (%+v) received (%+v) ", expectedStats, stats)
+	}
+
+	stats, token, err = spannerClient.ListChromiumDailyUsageStatsForFeatureID(
+		ctx,
+		"feature2",
+		time.Date(2000, time.January, 1, 0, 0, 0, 0, time.UTC),
+		time.Date(2000, time.December, 1, 0, 0, 0, 0, time.UTC),
+		2,
+		token,
+	)
+	if !errors.Is(err, nil) {
+		t.Errorf("expected no error during listing of metrics. received %s", err.Error())
+	}
+	if token != nil {
+		t.Error("expected no token")
+	}
+
+	expectedStats = []ChromiumDailyUsageStatWithDate{
+		{
+			Date: civil.Date{
+				Year:  2000,
+				Month: time.January,
+				Day:   1,
+			},
+			Usage: big.NewRat(89, 100),
+		},
+	}
+	if !reflect.DeepEqual(expectedStats, stats) {
+		t.Errorf("unequal stats. expected (%+v) received (%+v) ", expectedStats, stats)
+	}
+}
+
+func TestListChromiumDailyUsageStatsForFeatureID(t *testing.T) {
+	restartDatabaseContainer(t)
+	ctx := context.Background()
+	// Load up features.
+	sampleFeatures := getSampleFeatures()
+	webFeatureKeyToInternalFeatureID := map[string]string{}
+	for _, feature := range sampleFeatures {
+		id, err := spannerClient.UpsertWebFeature(ctx, feature)
+		if err != nil {
+			t.Errorf("unexpected error during insert of features. %s", err.Error())
+		}
+		webFeatureKeyToInternalFeatureID[feature.FeatureKey] = *id
+	}
+	addSampleChromiumUsageMetricsData(ctx, spannerClient, t, webFeatureKeyToInternalFeatureID)
```

Comment:
I reused these functions from `feature_search_test.go`, and I'm not sure if this might cause coupling issues. Is it okay to do this, or do I need to reimplement?

Reply:
I wouldn't mind refactoring these functions to take test data as input so they can be reused (to help with DRY) - do you think this is a good idea, and is there a good place where these functions might live?

Reply:
I would move each method to be in the `_test.go` variant of its main Upsert/Insert go file. For example:

`addSampleWebFeatureChromiumHistogramEnumValues` calls UpsertWebFeatureChromiumHistogramEnumValue. That method is in web_feature_chromium_histograms.go. So put it in web_feature_chromium_histograms_test.go (which I see createSampleWebFeatureChromiumHistogramEnums. So maybe you could refactor that and  addSampleWebFeatureChromiumHistogramEnumValues into one that only takes in sample data).


---
Also, I know the data from one table is very connected to data in another table. If you see it adding too much abstraction, feel free to punt on this.

Reply:
I will do so in a separate PR if that will suffice 🙂

Reply:
That works!! And feel free to do it after the milestone is complete. Just log an issue with the tech-debt tag